Python/specialize.c

```diff
@@ -383,8 +386,8 @@ _Py_PrintSpecializationStats(int to_file)
             hex_name[2*i+1] = Py_hexdigits[(rand[i]>>4)&15];
         }
         hex_name[40] = '\0';
-        char buf[64];
-        assert(strlen(dirname) + 40 + strlen(".txt") < 64);
+        char buf[128];
+        assert(strlen(dirname) + 40 + strlen(".txt") < 128);
```

Comment:
This is a bit scary, given that we're getting an arbitrarily long string from the environment variable. An assertion failure isn't a good experience on debug builds, and it just overflows the buffer on non-debug builds.

Not sure what the best solution is here, though.

Reply:
Maybe we could read and validate it once at startup, rather than lazily as stats are dumped?

Reply:
Yeah, I suppose validating at startup (where we *could* raise an exception) seems like the best solution.  Though this is enough of an internal developer detail that falling back to outputting to stderr (as already happens when the directory doesn't exist) might also be ok.